src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -174,6 +174,8 @@
     (.close ds))
   (when-let [ds (get-in context [:shared-globals :scf-read-db :datasource])]
     (.close ds))
+  (when-let [pool (:job-pool context)]
+    (stop-and-reset-pool! pool))
```

Comment:
I think this will probably need to be applied to the sync code in extensions as well